### PR TITLE
Add Trino 385 release notes

### LINKED ISSUE #12644: Release notes for 384

### LINKED ISSUE #12686: Release notes for 385

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 385 release.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Docs only.

> How would you describe this change to a non-technical end user or system administrator?

Add the release notes for Trino 385 with sections for general, security, connectors and other topics as included in the release.

## Related issues, pull requests, and links

* Fixes #12686 

## Documentation

( ) No documentation is needed.
(✅) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅) No release notes entries required.
( ) Release notes entries required with the following suggested text:

## Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 3 June 2022 ✅ 
62f83ee943092f584862ac46318af910c4c4d889 @dain #12414 ✅ rn ✅ docs

## 4 June 2022 ✅ 
bf19f47c0e432277149a6549f2224ffcde9dd7e2 @findepi #12579 ✅ rn ✅ docs
651c5422d8033df68ff11601e11d8ccf000fa650 @findepi #12579 ✅ rn ✅ docs

## 5 June 2022 ✅ 
db5db23de77e33229a362d78eeac11c05c7ee13f @electrum #11871 ✅ rn ✅ docs
af41e15020f88ef516e8fbeb5a1e54064cb0d904 @electrum #11871 ✅ rn ✅ docs
4b38b19947976c5a91b5a33e69899e08c5c22359 @findepi #12690 ✅ rn ✅ docs
1f634b72e0f09f5079c391723d13caedab7501ff @ebyhr #12642 ✅ rn ✅ docs
c37931c1fd4efb60648694fc57ba974844c385f6 @ebyhr #12679 ✅ rn ✅ docs

## 6 June 2022 ✅ 
a8758737e0a8f1e5ac98b14a72e76e87f2af09a9 @ebyhr #12695 ✅ rn ✅ docs
f9c25e278ff9ac069fb6e558102c5c2b2ad8b8e1 @findepi #12696 ✅ rn ✅ docs
d8f2b6d0d81f3902fe4cc86963a5b4ae735e422d @findepi #12522 ✅ rn ✅ docs
6ec04ffca16fbcaa848c988c14cd0ff06684d817 @findepi #12522 ✅ rn ✅ docs
34e3412eafee6341ee30357394e60b58b4b9b931 @findepi #12673 ✅ rn ✅ docs
1214981c1e28cb8519606140c99e00e515491b39 @kasiafi #12049 ✅ rn ✅ docs
605ef9541e9d5dc49308e24cd32f5353e0178fbf @kasiafi #12049 ✅ rn ✅ docs
3c826168d9853406c85f7ac35941ae90abe7ce09 @kasiafi #12049 ✅ rn ✅ docs
0ebe8c3865aa130d48480ccd977d8aa69206a830 @kasiafi #12049 ✅ rn ✅ docs
497a247ffd5658283e5cebb01976ea2455da93eb @findepi #12699 ✅ rn ✅ docs
fff9862195351ed7e674386368bf9d1c9c17f28c @ebyhr #12455 ✅ rn ✅ docs

## 7 June 2022 ✅
3773873c2b42eb41a14400440a9bb81e320573e1 @findepi #12707 ✅ rn ✅ docs
56013a0e11929538df84168c8d6ce36e526ef7c8 @kokosing #12716 ✅ rn ✅ docs
08cf8e2c375a463f051f5e813c876be5cf534cd9 @ebyhr #12482 ✅ rn ✅ docs
801f7fe828308793b8d4a65b60a9fd5c0d4c1fe6 @findepi #12622 ✅ rn ✅ docs
4a0f84f7e9987a71e62a24faca4d90528e1be6e2 @findepi #10258 ✅ rn ✅ docs
5d939cb932e67046d5a584d49fbe7463be4dadb1 @findepi #12724 ✅ rn ✅ docs
c2f851705f462c74426ffe0ab96175fb8365ff41 @findepi #12614 ✅ rn ✅ docs
e757e67da2ffd78761be1ea8ab1fe520fe07b578 @findepi #12614 ✅ rn ✅ docs
47bc40aeb0c9ecfa377179e91a1d85a7078e898e @ebyhr #12715 ✅ rn ✅ docs
08187fcafc0ba8edf980c0af89029e5a4f67fd2e @ebyhr #12715 ✅ rn ✅ docs
5ac84049101fc79cba1ce867abcc100326c21467 @ebyhr #12606 ✅ rn ✅ docs

## 8 June 2022 ❌ 
216c8d54a7c0f7a5fb26963c74a5d07f488046de @kasiafi #12677 ✅ rn ✅ docs
fa0de5291f3adef80e37f496579a66bcb32b0c50 @findepi #12024 ✅ rn ✅ docs
f6ab1d6cf40b8bbfaaf8d1a7f65a57e6b47e0658 @findepi #12717 ✅ rn ✅ docs
f7710c390f6fd6ec76e3e7a9f2fdaa9004abecbe @losipiuk #12639 ✅ rn ✅ docs
66a5ef7af821f721651e8dfef2a2266ecd24ae4c @losipiuk #12639 ✅ rn ✅ docs
98945e8dc7239cf6aac87503106c0e73e72f0609 @losipiuk #12639 ✅ rn ✅ docs
f712f2ccc7946c3ca3218d18a5aebaa6dfb31779 @arhimondr #12721 ✅ rn ✅ docs
049ef8130ccc254d5fbbeab8fb9161e70683d6fa @arhimondr #12721 ✅ rn ✅ docs
e196b370a226d615717d1e065d0b68cee784ef54 @arhimondr #12721 ✅ rn ✅ docs
1afed4f85efbb9a0425f7385baa3c470a8a33d13 @findepi #12705 ✅ rn ✅ docs
ef22bc9b3f24b9fa31b61ce0bdb7713db58828bd @ebyhr #12437 ✅ rn ✅ docs
2218d9ab04c46dcd24b50c57cb56f52049f7edb8 @findepi #12567 ✅ rn ❌ docs
bd3b0ae84f1a04839e4cdc8a6fe523393d457289 @findepi #12567 ✅ rn ❌ docs
b3851853f5ae7659abbf4fc3370a5a3506a98169 @findepi #12671 ✅ rn ✅ docs
6b3efd4d2f92a4dd6f6db7d2d3b939e99b17e406 @findepi #12735 ✅ rn ✅ docs
103b9cac0ee5588ad96800ffec207dcc411ed7e1 @kasiafi #12682 ✅ rn ✅ docs
57c48519122af83e561736b8b2d9b7c69a9f8907 @martint #12703 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-385.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## Hive connector
+
+* Allow cancelling a query on a transactional table if it is waiting for a lock. ({issue}`11798`)
+* Propagate compression codec selected by `hive.compression-codec` config
+  property or `compression_codec` session property to Avro file writer. ({issue}`12639`)
+
+## Iceberg connector
+
+* Improve query performance when a table consists of many small files. ({issue}`12579`)
+* Improve query performance when performing a delete or update. ({issue}`12671`)
```
Comment (colebow) [inline docs/src/main/sphinx/release/release-385.md:27]: @findepi is this reasonable?

File: docs/src/main/sphinx/release/release-385.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+* Add support for time travel. ({issue}`10258`)
```
Comment (colebow) [inline docs/src/main/sphinx/release/release-385.md:28]: I think I'd like to document what time travel is and reference that here, rather than try to explain how to use time travel in release notes. Going to get a PR for this up ASAP.

Comment (martint) [reply docs/src/main/sphinx/release/release-385.md:28]: I would rephrase as "Add support for [VERSION | TIMESTAMP] AS OF clause" or something similar. There's not such a thing as "time travel" in the SQL language or spec.

Also, we already support querying older snapshots of a table by referencing special table names.

File: docs/src/main/sphinx/release/release-385.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+* Show Iceberg location and `format_version` in `SHOW CREATE MATERIALIZED VIEW`. ({issue}`12504`)
+
+## MariaDB connector
+
+* Add support for `timestamp` type. ({issue}`12200`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-385.md:33]: The type is actually `timestamp(p)`

File: docs/src/main/sphinx/release/release-385.md (hunk 2 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-385.md:28]: I don't see anything there that can be linked to...

File: docs/src/main/sphinx/release/release-385.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Delta Lake connector
+
+* Add support for the [appendOnly field](https://docs.delta.io/latest/delta-batch.html#-table-properties). ({issue}`12635`)
+* Add support for column comments when creating a table or a column. ({issue}`12455`, {issue}`12715`))
+
+## Hive connector
+
+* Allow cancelling a query on a transactional table if it is waiting for a lock. ({issue}`11798`)
+* Support compression codec configuration for Avro file writing with the
```
Comment (martint) [inline docs/src/main/sphinx/release/release-385.md:21]: Add support for selecting a compression scheme when writing Avro files via the ....

**Review by mosabua — APPROVED** [review]
Ship it!

**Comment by martint** [comment]
Merged as 1a14cf87d2970ee35ee9af58f425057cea054090

File: docs/src/main/sphinx/release/release-385.md (hunk 1 above)
Comment (findepi) [reply docs/src/main/sphinx/release/release-385.md:28]: > Improve query performance when performing a delete or update

not really. the delete/update performance is the same as it used to be
however, delete/update now produces files that are more efficient to query later
